pkg/kubelet/container_bridge.go

```diff
@@ -43,15 +43,16 @@ func createCBR0(wantCIDR *net.IPNet) error {
 		glog.Error(err)
 		return err
 	}
-	// restart docker
+	// Stop docker so that babysitter process can restart it again with proper configurations and
+	// checkpoint file
 	// For now just log the error. The containerRuntime check will catch docker failures.
 	// TODO (dawnchen) figure out what we should do for rkt here.
 	if util.UsingSystemdInitSystem() {
-		if err := exec.Command("systemctl", "restart", "docker").Run(); err != nil {
+		if err := exec.Command("systemctl", "stop", "docker").Run(); err != nil {
```

Comment:
We require babysit daemons on KNode since 1.0 release: https://github.com/kubernetes/kubernetes/issues/2884
It has nothing to do with salt. Daemons could be dead due to various reasons, babysitter process just make sure it always run happily. Those "modern" init system, like systemd and upstart which can do health-check, not a simple liveness check, take over the entire babysit work for daemons: kubelet and docker. 

For example, among our cloud providers, unbuntu trusty, we are using upstart to babysit daemons; on coreos, we are using systemd to babysit daemons already. We used to babysit kube-proxy too until now it run as daemon-set, and is babysitted by kubelet directly. 

On containerVM image today, the init system can not do health-check by itself, we introduced monit first, later supervisord to help with this as the extension of init system on KNode. 

--configure-cbr0 is a hacky way we introduced as a trade-off back then. The only reason we call docker service restart directly is monit used to take 2mins (default config) to performan healthcheck, but changing the default config of monit requires more salt operations, which leads other issues. This change is simply reset this back to the initial state. Actually people did ask if we can simply stop docker here without start to cleanup our bootup sequence before. 

If we let only init sytem to manage the lifecycle of daemons on KNode, flock can be introduced to docker-checker.sh script, but it is not that necessary.